ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+
+
+/**
+ * @param resource $stream_id
+ * @todo: should $file be `resouce|string`? it looks like it tries to accept anything?
+ */
+function imap_savebody($stream_id, $file, int $msg_no, string $section = '', int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchheader($stream_id, int $msg_no, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_fetchstructure($stream_id, int $msg_no, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_gc($stream_id, int $flags): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_expunge($stream_id): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_delete($stream_id, int $msg_no, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_undelete($stream_id, int $msg_no, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_check($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_listscan($stream_id, string $ref, string $pattern, string $content) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_mail_copy($stream_id, string $msglist, string $mailbox, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_mail_move($stream_id, string $sequence, string $mailbox, int $options = 0): bool {}
+
+/**
+ * @return string|false
+ */
+function imap_mail_compose(array $envelope, array $body) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_createmailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_renamemailbox($stream_id, string $old_name, string $new_name): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_deletemailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_subscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_unsubscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_append($stream_id, string $folder, string $message, string $options = '', string $internal_date = ''): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_ping($stream_id): bool {}
+
+/**
+ * @return string|false
+ */
+function imap_base64(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_qprint(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_8bit(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_binary(string $text) {}
+
+
+function imap_utf8(string $mime_encoded_text): string {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_status($stream_id, string $mailbox, int $options) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
```

Comment:
I don't think that `false` can actually be returned (failing zend_fetch_resource() throws).

Reply:
Ok, I asked about this in R11 but no update yet - is `object` or `stdClass` a better return type for those that return one - I can see arguments either way, so I'm just hoping there's some existing standard for to follow?